### ISSUE TITLE
Formatting suddenly stopped working

### ISSUE THREAD

**Issue by chris-martin** [haskell/haskell-language-server#3667]
I have no idea what changed. The autoformat command was working one day, stopped working the next.

When I run the format command in vscode, I see this in the Haskell output tab:

```
2023-06-22T18:35:59.131211Z | Warning | No plugin enabled for STextDocumentFormatting
[Warn  - 12:35:59 PM] No plugin enabled for STextDocumentFormatting
[Error - 12:35:59 PM] Request textDocument/formatting failed.
  Message: No plugin enabled for STextDocumentFormatting, available: cabalfmt
  Code: -32600 
```

An example environment can be found here - https://github.com/typeclasses/integer-types/tree/b8ec4030f0e7ca0160b4f83b8334bd21a7938855 - Nix+direnv will install haskell-language-server et al.

My "formatting provider" is set to "ormolu." Ormolu is on the PATH. I notice that the setting description includes has a note says "Ensure the plugin is enabled." but it doesn't say what plugin to enable.

**Comment by fendor** [comment]
Since only hls-cabal-fmt-plugin is enabled, you have probably upgraded your HLS version to a version where none of the plugins are either installed or available.
So, what's the HLS and GHC version? Should be at the top of the logs.

**Comment by chris-martin** [comment]
`haskell-language-server version: 1.10.0.0 (GHC: 9.6.2)`

**Comment by fendor** [comment]
Apparently, no formatter compiles or is compatible with that GHC version, so we can't ship any either. https://haskell-language-server.readthedocs.io/en/latest/support/plugin-support.html#current-plugin-support-tiers

**Comment by chris-martin** [comment]
Is there any way I can just get it to run the ormolu that's already installed? I can run `ormolu` from the command line and it's fine, only HLS has a problem with it.

**Comment by fendor** [comment]
No, currently, for ormolu, I think we only support built-in ormolu versions. I think you can do it with fourmolu. Also, maybe building https://github.com/haskell/haskell-language-server/pull/3668 from source might be helpful.

**Comment by michaelpj** [comment]
Unfortunately, although the fourmolu plugin got fixed for 9.6, it wasn't enabled so I don't think it's in the latest release either.

I'm also working on getting stylish-haskell updated: https://github.com/haskell/stylish-haskell/pull/459
